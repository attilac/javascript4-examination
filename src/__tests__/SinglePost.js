import React from 'react';
import { mount } from 'enzyme';
import SinglePost from '../components/SinglePost';

describe('singlepost unit tests', ()=>{
  const onclick = jest.fn();
  const wrapper = mount(<SinglePost title="testTitle" content="testContent" author="testAuthor" id="1" date="2018-1-23 14:44:38" currentPersona="testAuthor" onClick={onclick} />)
  const noBtnWrapper = mount(<SinglePost title="testTitle" content="testContent" author="testAuthor" id="1" date="2018-1-23 14:44:38" currentPersona="anotherPersona" onClick={onclick} />)
  
  it('should render button when author and currentPersona match', ()=>{
    expect(wrapper.find('article').html()).toContain('<button')
  });

  it('should call onclick with id', ()=>{
    wrapper.find('button').simulate('click');
    expect(onclick).toHaveBeenCalledWith('1');
  });

  it('should not render button when author and currentPersona mismatch', ()=> {
    expect(noBtnWrapper.find('article').html()).not.toContain('<button');
  });

  it('should render title & content on page', ()=> {
    expect(wrapper.find('article h2').text()).toEqual('testTitle');
    expect(wrapper.find('ReactMarkdown').text()).toEqual('testContent');
  })

  it('should match snapshots', ()=> {
    expect(wrapper).toMatchSnapshot();
    expect(noBtnWrapper).toMatchSnapshot();
  })

  it('should console errors when sent no props', ()=> {
    let error;
    const warn = jest.spyOn(global.console, 'error').mockImplementation((e) => {
      error = e;
      return true;
    });;
    mount(<SinglePost />)
    expect(warn).toHaveBeenCalled();
    expect(error).toContain('Warning');
    warn.mockReset();
    warn.mockRestore();
  })

});
