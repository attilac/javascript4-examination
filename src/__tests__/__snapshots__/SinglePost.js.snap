// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`singlepost unit tests should match snapshots 1`] = `
<SinglePost
  author="testAuthor"
  content="testContent"
  currentPersona="testAuthor"
  date="2018-1-23 14:44:38"
  id="1"
  onClick={[Function]}
  title="testTitle"
>
  <div
    className="w-full shadow p-6 m-6 border rounded relative"
  >
    <article
      className="mb-2"
    >
      <Button
        className="absolute pin-t pin-r mx-4 my-6"
        danger={true}
        onClick={[Function]}
      >
        <button
          className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r mx-4 my-6"
          data-test="button"
          onClick={[Function]}
        >
          x
        </button>
      </Button>
      <h2
        className="mb-4 w-4/5"
      >
        testTitle
      </h2>
      <ReactMarkdown
        className="mb-4 text-grey-darkest markdown"
        escapeHtml={true}
        renderers={Object {}}
        skipHtml={false}
        source="testContent"
        transformLinkUri={[Function]}
      >
        <div
          className="mb-4 text-grey-darkest markdown"
          key="root-1-1"
        >
          <p
            key="paragraph-1-1"
          >
            testContent
          </p>
        </div>
      </ReactMarkdown>
      <p
        className="text-grey-dark mb-4"
      >
        Posted by: 
        testAuthor
         @ 
        2018-1-23 14:44:38
      </p>
    </article>
    <Comments
      currentPersona="testAuthor"
      postId="1"
    >
      <div
        className="py-2"
      >
        <h2
          className="text-indigo-darker border-b mb-4"
        >
          Comments
        </h2>
        <CreateNewComment
          author="testAuthor"
          postId="1"
          updateComments={[Function]}
        >
          <form
            className="container mx-auto flex flex-col p-6"
            onSubmit={[Function]}
          >
            <label
              className="mb-2"
              htmlFor="comment"
            >
              Comment
            </label>
            <textarea
              className="shadow rounded border mb-2 p-2"
              id="comment"
              name="comment"
              onChange={[Function]}
              value=""
            />
            <input
              className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
              type="submit"
              value="Comment"
            />
          </form>
        </CreateNewComment>
      </div>
    </Comments>
  </div>
</SinglePost>
`;

exports[`singlepost unit tests should match snapshots 2`] = `
<SinglePost
  author="testAuthor"
  content="testContent"
  currentPersona="anotherPersona"
  date="2018-1-23 14:44:38"
  id="1"
  onClick={[Function]}
  title="testTitle"
>
  <div
    className="w-full shadow p-6 m-6 border rounded relative"
  >
    <article
      className="mb-2"
    >
      <h2
        className="mb-4 w-4/5"
      >
        testTitle
      </h2>
      <ReactMarkdown
        className="mb-4 text-grey-darkest markdown"
        escapeHtml={true}
        renderers={Object {}}
        skipHtml={false}
        source="testContent"
        transformLinkUri={[Function]}
      >
        <div
          className="mb-4 text-grey-darkest markdown"
          key="root-1-1"
        >
          <p
            key="paragraph-1-1"
          >
            testContent
          </p>
        </div>
      </ReactMarkdown>
      <p
        className="text-grey-dark mb-4"
      >
        Posted by: 
        testAuthor
         @ 
        2018-1-23 14:44:38
      </p>
    </article>
    <Comments
      currentPersona="anotherPersona"
      postId="1"
    >
      <div
        className="py-2"
      >
        <h2
          className="text-indigo-darker border-b mb-4"
        >
          Comments
        </h2>
        <CreateNewComment
          author="anotherPersona"
          postId="1"
          updateComments={[Function]}
        >
          <form
            className="container mx-auto flex flex-col p-6"
            onSubmit={[Function]}
          >
            <label
              className="mb-2"
              htmlFor="comment"
            >
              Comment
            </label>
            <textarea
              className="shadow rounded border mb-2 p-2"
              id="comment"
              name="comment"
              onChange={[Function]}
              value=""
            />
            <input
              className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
              type="submit"
              value="Comment"
            />
          </form>
        </CreateNewComment>
      </div>
    </Comments>
  </div>
</SinglePost>
`;
